packages/toolkit/core-kit/src/password-policy.ts

```diff
@@ -0,0 +1,317 @@
+      issues.push({
+        code: 'password_rejected.too_short',
+      });
+    } else if (password.length > this.policy.length.max) {
+      issues.push({
+        code: 'password_rejected.too_long',
+      });
+    }
+
+    const characterTypes = this.checkCharTypes(password);
+    if (characterTypes === 'unsupported') {
+      issues.push({
+        code: 'password_rejected.unsupported_characters',
+      });
+    } else if (!characterTypes) {
+      issues.push({
+        code: 'password_rejected.character_types',
+        interpolation: { min: this.policy.characterTypes.min },
+      });
+    }
+
+    if (this.policy.rejects.pwned && (await this.hasBeenPwned(password))) {
+      issues.push({
+        code: 'password_rejected.pwned',
+      });
+    }
+
+    if (this.policy.rejects.repetitionAndSequence) {
+      if (this.hasRepetition(password)) {
+        issues.push({
+          code: 'password_rejected.repetition',
+        });
+      }
+
+      if (this.hasSequentialChars(password)) {
+        issues.push({
+          code: 'password_rejected.sequence',
+        });
+      }
+    }
+
+    issues.push(
+      ...this.hasWords(password).map<PasswordIssue>(({ type, value }) => ({
+        code: 'password_rejected.restricted_words',
+        interpolation: { type, value },
+      }))
+    );
+
+    return issues;
+  }
+  /* eslint-enable @silverhand/fp/no-mutating-methods */
+
+  /**
+   * Check if the given password contains enough character types.
+   *
+   * @param password - Password to check.
+   * @returns Whether the password contains enough character types; or `'unsupported'`
+   * if the password contains unsupported characters.
+   */
+  checkCharTypes(password: string): boolean | 'unsupported' {
+    const characterTypes = new Set<string>();
+    for (const char of password) {
+      if (char >= 'a' && char <= 'z') {
+        characterTypes.add('lowercase');
+      } else if (char >= 'A' && char <= 'Z') {
+        characterTypes.add('uppercase');
+      } else if (char >= '0' && char <= '9') {
+        characterTypes.add('digits');
+      } else if (PasswordPolicyChecker.symbols.includes(char)) {
+        characterTypes.add('symbols');
+      } else {
+        return 'unsupported';
+      }
+    }
+
+    return characterTypes.size >= this.policy.characterTypes.min;
+  }
+
+  /**
+   * Check if the given password has been pwned.
+   *
+   * @param password - Password to check.
+   * @returns Whether the password has been pwned.
+   */
+  async hasBeenPwned(password: string): Promise<boolean> {
+    const hash = await crypto.subtle.digest('SHA-1', new TextEncoder().encode(password));
+    const hashHex = Array.from(new Uint8Array(hash))
+      .map((binary) => binary.toString(16).padStart(2, '0'))
+      .join('');
+    const hashPrefix = hashHex.slice(0, 5);
+    const hashSuffix = hashHex.slice(5);
+    const response = await fetch(`https://api.haveibeenpwned.com/range/${hashPrefix}`);
```

Comment:
Should add a fallback logic? Incase the connection is not stable.

Reply:
thought about this, but if it's enabled it looks not good to fall back without error

Reply:
I was thinking maybe log the error, and let the end user moving forward.

Reply:
posted this in Slack, will update if needed